SharedSource/Development/Moq/UnitTesting/DecoratorTestHelper.cs

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+//
+// See the NOTICE file distributed with this work for additional information
+// regarding copyright ownership.  rubicon licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may not use this
+// file except in compliance with the License.  You may obtain a copy of the
+// License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+// License for the specific language governing permissions and limitations
+// under the License.
+//
+using System;
+using System.Linq.Expressions;
+using Moq;
+using NUnit.Framework;
+using Remotion.Utilities;
+
+// ReSharper disable once CheckNamespace
+namespace Remotion.Development.Moq.UnitTesting
+{
+  /// <summary>
+  /// Provides functionality for testing decorator methods that do nothing else but forward to the equivalent methods on a decorated object.
+  /// </summary>
+  /// <typeparam name="TInterface">The type of the interface.</typeparam>
+  public partial class DecoratorTestHelper<TInterface>
+      where TInterface : class
+  {
+    private readonly TInterface _decorator;
+    private Mock<TInterface> _decoratedMock;
+
+    public DecoratorTestHelper (TInterface decorator, Mock<TInterface> decoratedMock)
+    {
+      ArgumentUtility.CheckNotNull ("decorator", decorator);

```

Comment:
replace argument checks with if/throw. This, too, is best practice for the shared source files to prevent dependencies to argumentutility where possible/easy. Sometimes, this was forgotten and then also forgotten in the nuspec file...

Reply:
?